### PR TITLE
token-assert:Make sure newlines get fixed

### PR DESCRIPTION
**Pull request by gero3**
The new lines rules should now be able to get fixed too.

### CODE REVIEW

**Comment by mikesherov** [comment]
@gero3, good work! Just needs more tests, one for each of the conditions.

File: test/token-assert.js [hunk 1]
```diff
@@ -564,6 +564,40 @@ describe('modules/token-assert', function() {
 
                 assert(!onError.called);
             });
+
+            it('should edit the whitespaceBefore', function() {
+                var file = createJsFile('x\n=y;');
+
+                var tokenAssert = new TokenAssert(file);
+                var onError = sinon.spy();
```
Comment (mikesherov) [inline test/token-assert.js]: Is the spy needed in each of these tests? If not, please remove. Thanks!

File: lib/string-checker.js [hunk 2]
```diff
@@ -238,7 +238,6 @@ StringChecker.prototype = {
 
                 source = file.render();
                 sourceTree = JsFile.parse(source, this._esprima, this._configuration.getEsprimaOptions());
-
```
Comment (mikesherov) [inline lib/string-checker.js]: Remove this unnecessary whitespace change. Thanks!

**Comment by mikesherov** [comment]
Thanks again @gero3!

File: test/token-assert.js (hunk 1 above)
Comment (gero3) [inline test/token-assert.js]: fixed

File: test/token-assert.js [hunk 3]
```diff
@@ -564,6 +564,38 @@ describe('modules/token-assert', function() {
 
                 assert(!onError.called);
             });
+
+            it('should edit the whitespaceBefore', function() {
+                var file = createJsFile('x\n=y;');
+
+                var tokenAssert = new TokenAssert(file);
+                tokenAssert.on('error', function() {});
```
Comment (mikesherov) [inline test/token-assert.js]: I should have been more clear. Are the event listeners needed either? Remove this line in each test if not.

Comment (gero3) [inline test/token-assert.js]: It throws an exception if I don't listen to the event. That is why I'm simply adding the function.

> TypeError: Uncaught, unspecified "error" event.

Comment (mikesherov) [inline test/token-assert.js]: Ok. Thanks!

**Comment by mikesherov** [comment]
I ended up rewriting the tests because there were a few errors in them. Thanks again @gero3
